libvast/src/expression_visitors.cpp

```diff
@@ -248,15 +250,24 @@ caf::expected<void> validator::operator()(const predicate& p) {
 
 caf::expected<void> validator::
 operator()(const attribute_extractor& ex, const data& d) {
-  if (ex.attr == system::type_atom::get_value()
-      && !(caf::holds_alternative<std::string>(d)
-           || caf::holds_alternative<pattern>(d)))
-    return make_error(ec::syntax_error,
-                      "type attribute extractor requires string or pattern "
-                      "operand",
-                      ex.attr, op_, d);
-  else if (ex.attr == system::timestamp_atom::get_value()
-           && !caf::holds_alternative<time>(d))
+  if (ex.attr == system::type_atom::get_value()) {
```

Comment:
It looks like only expressions of the form `#type op RHS` get checked. But we want to validate *all* pattern occurrences, that is, every predicate operand that is a pattern should be checked.

Reply:
This should be all the places now.